### ISSUE TITLE
TypeError: Cannot convert undefined or null to object

### ISSUE THREAD

**Issue by brandon-bethke-neudesic** [olivernn/lunr.js#360]
The following code will throw an exception: TypeError: Cannot convert undefined or null to object

```
const lunr = require("lunr");
let lunarbuilder;
lunr(builder => {
    lunrbuilder = builder;
    builder.field('name');
    builder.field('description');
    builder.ref('id');
});

lunrbuilder.add({
    name: 'myname',
    description: 'mydescription',
    type: 'core:test',
    key: '1',
    id: 1,
    url: '/api/graph/core:test/1',
    createdAt: '2018-07-19T20:33:18.964Z',
    archived: false
});

lunrbuilder.add({
    name: 'myname2',
    description: 'mydescription',
    type: 'core:test',
    key: '2',
    id: 2,
    url: '/api/graph/core:test/2',
    createdAt: '2018-07-19T20:33:18.968Z',
    archived: false
});

let search = lunrbuilder.build();
search.search("+name:myname +description:value");
```

**Comment by olivernn** [comment]
Thanks for reporting, I was able to [reproduce](https://jsfiddle.net/qza1ho7b/1/) this issue. It looks like it is triggered when the search contains two required terms each scoped by a field but one that matches nothing.

I've tracked the bug down to the way that `lunr.Set.complete` and `lunr.Set.empty` are handled in `lunr.Set#union` and `lunr.Set#intersect`. I'll put together a bug fix and a new release in the next few days.
